// cspell-lib/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.clearCachedFiles = exports.clearCachedSettings = exports.ExclusionHelper = exports.Text = void 0;
__exportStar(require("cspell-io"), exports);
var validator_1 = require("./validator");
Object.defineProperty(exports, "checkText", { enumerable: true, get: function () { return validator_1.checkText; } });
Object.defineProperty(exports, "IncludeExcludeFlag", { enumerable: true, get: function () { return validator_1.IncludeExcludeFlag; } });
Object.defineProperty(exports, "validateText", { enumerable: true, get: function () { return validator_1.validateText; } });
var Settings_1 = require("./Settings");
Object.defineProperty(exports, "defaultSettingsFilename", { enumerable: true, get: function () { return Settings_1.defaultFileName; } });
__exportStar(require("./Settings"), exports);
var SpellingDictionary_1 = require("./SpellingDictionary");
Object.defineProperty(exports, "CompoundWordsMethod", { enumerable: true, get: function () { return SpellingDictionary_1.CompoundWordsMethod; } });
Object.defineProperty(exports, "createSpellingDictionary", { enumerable: true, get: function () { return SpellingDictionary_1.createSpellingDictionary; } });
Object.defineProperty(exports, "getDictionary", { enumerable: true, get: function () { return SpellingDictionary_1.getDictionary; } });
Object.defineProperty(exports, "refreshDictionaryCache", { enumerable: true, get: function () { return SpellingDictionary_1.refreshDictionaryCache; } });
var TextDocumentSettings_1 = require("./Settings/TextDocumentSettings");
Object.defineProperty(exports, "combineTextAndLanguageSettings", { enumerable: true, get: function () { return TextDocumentSettings_1.combineTextAndLanguageSettings; } });
var TextDocumentSettings_2 = require("./Settings/TextDocumentSettings");
Object.defineProperty(exports, "constructSettingsForText", { enumerable: true, get: function () { return TextDocumentSettings_2.combineTextAndLanguageSettings; } });
const Text = require("./util/text");
exports.Text = Text;
const ExclusionHelper = require("./exclusionHelper");
exports.ExclusionHelper = ExclusionHelper;
var LanguageIds_1 = require("./LanguageIds");
Object.defineProperty(exports, "getLanguagesForExt", { enumerable: true, get: function () { return LanguageIds_1.getLanguagesForExt; } });
__exportStar(require("./trace"), exports);
const Settings_2 = require("./Settings");
const SpellingDictionary_2 = require("./SpellingDictionary");
async function clearCachedSettings() {
    await Promise.all([
        Settings_2.clearCachedFiles(),
        SpellingDictionary_2.refreshDictionaryCache(0),
    ]);
}
exports.clearCachedSettings = clearCachedSettings;
exports.clearCachedFiles = clearCachedSettings;
